web/pages/[username]/[contractSlug].tsx

```diff
@@ -119,6 +122,17 @@ export function ContractPageContent(props: FirstArgument<typeof ContractPage>) {
 
   return (
     <Page rightSidebar={rightSidebar}>
+      {user &&
+        contract.creatorId === user.id &&
+        Date.now() - contract.createdTime < 10 * 1000 && (
```

Comment:
After react renders, it won't necessarily know to render again when this condition changes. A more robust alternative would be to set a timeout that calls a set state. E.g. https://usehooks-ts.com/react-hook/use-timeout

Reply:
Hmmm the logic should check against a separate time, not the time since it loaded.